parquet-generator/src/main/resources/ByteBitPackingVectorLE

```diff
@@ -0,0 +1,3218 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.column.values.bitpacking;
+
+import jdk.incubator.vector.*;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This is an auto-generated source file and should not edit it directly.
+ */
+public abstract class ByteBitPackingVectorLE {
+  private static final BytePacker[] packers = new BytePacker[33];
+
+  static {
+    packers[0] = new Packer0();
+    packers[1] = new Packer1();
+    packers[2] = new Packer2();
+    packers[3] = new Packer3();
+    packers[4] = new Packer4();
+    packers[5] = new Packer5();
+    packers[6] = new Packer6();
+    packers[7] = new Packer7();
+    packers[8] = new Packer8();
+    packers[9] = new Packer9();
+    packers[10] = new Packer10();
+    packers[11] = new Packer11();
+    packers[12] = new Packer12();
+    packers[13] = new Packer13();
+    packers[14] = new Packer14();
+    packers[15] = new Packer15();
+    packers[16] = new Packer16();
+    packers[17] = new Packer17();
+    packers[18] = new Packer18();
+    packers[19] = new Packer19();
+    packers[20] = new Packer20();
+    packers[21] = new Packer21();
+    packers[22] = new Packer22();
+    packers[23] = new Packer23();
+    packers[24] = new Packer24();
+    packers[25] = new Packer25();
+    packers[26] = new Packer26();
+    packers[27] = new Packer27();
+    packers[28] = new Packer28();
+    packers[29] = new Packer29();
+    packers[30] = new Packer30();
+    packers[31] = new Packer31();
+    packers[32] = new Packer32();
+  }
+
+  public static final BytePackerFactory factory = new BytePackerFactory() {
+    public BytePacker newBytePacker(int bitWidth) {
+      return packers[bitWidth];
+    }
+  };
+
+  private static final class Packer0 extends BytePacker {
+    private int unpackCount = 0;
+
+    private Packer0() {
+      super(0);
+    }
+
+    public int getUnpackCount() {
+      return unpackCount;
+    }
+
+    public final void pack8Values(final int[] in, final int inPos, final byte[] out, final int outPos) {
+    }
+
+    public final void pack32Values(final int[] in, final int inPos, final byte[] out, final int outPos) {
+    }
+
+    public final void unpack8Values(final byte[] in, final int inPos, final int[] out, final int outPos) {
+    }
+
+    public final void unpack8Values(final ByteBuffer in, final int inPos, final int[] out, final int outPos) {
+    }
+
+    public final void unpack32Values(final byte[] in, final int inPos, final int[] out, final int outPos) {
+    }
+
+    public final void unpack32Values(final ByteBuffer in, final int inPos, final int[] out, final int outPos) {
+    }
+
+    public final void unpackValuesVector(final byte[] input, final int inPos, final int[] output, final int outPos) {
+    }
+
+    public final void unpackValuesVector(final ByteBuffer input, final int inPos, final int[] output, final int outPos) {
```

Comment:
All these empty definitions can be removed if we introduce a new class ByteVectorPacker which inherit from existing BytePacker.

Reply:
We also need to ensure that we invoke both scalar generators and vector generators for jdk17 profile.